src/otx/core/data/pre_filtering.py

```diff
@@ -15,21 +16,35 @@
     from datumaro.components.dataset_base import DatasetItem
 
 
-def pre_filtering(dataset: DmDataset, data_format: str) -> DmDataset:
-    """Filtering invalid data in datumaro dataset."""
-    dataset = DmDataset.filter(dataset, is_non_empty_item)
-    dataset = DmDataset.filter(dataset, is_valid_annot, filter_annotations=True)
+def pre_filtering(dataset: DmDataset, data_format: str, unannotated_bg_ratio: float) -> DmDataset:
+    """Pre-filtering function to filter the dataset based on certain criteria.
 
-    return remove_unused_labels(dataset, data_format)
+    Args:
+        dataset (DmDataset): The input dataset to be filtered.
+        data_format (str): The format of the dataset.
+        unannotated_bg_ratio (float): The ratio of background unannotated items to be used.
+            This must be a float between 0 and 1.
 
+    Returns:
+        DmDataset: The filtered dataset.
 
-def is_non_empty_item(item: DatasetItem) -> bool:
-    """Return whether DatasetItem's annotation is non-empty."""
-    if not (item.subset == "train" and len(item.annotations) == 0):
-        return True
-    msg = "There are empty annotation items in train set, they will be filtered out before training."
+    Raises:
+        None.
+    """
+    used_background_items = []
+    msg = f"There are empty annotation items in train set, Of these, only {unannotated_bg_ratio*100}% are used."
     warnings.warn(msg, stacklevel=2)
-    return False
+    if unannotated_bg_ratio > 0:
+        empty_items = [item for item in dataset if item.subset == "train" and len(item.annotations) == 0]
+        used_background_items = sample(empty_items, int(len(empty_items) * unannotated_bg_ratio))
```

Comment:
```suggestion
    if unannotated_bg_ratio > 0:
        empty_item_ids = [item.id for item in dataset if item.subset == "train" and len(item.annotations) == 0]
        used_background_ids = set(sample(empty_items, int(len(empty_items) * unannotated_bg_ratio)))
```
1. Checking `DatasetItem` equivalence is heavier than `str`'s.
2. `item in list` is a O(n) algorithm.

Reply:
Thanks for letting me know what I missed. - https://github.com/openvinotoolkit/training_extensions/pull/2965/commits/ba350f2b7cbedf68e7306874b7e02a7ec36a95e4